intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/normalizedcache/DatabaseNormalizedCacheProvider.kt

```diff
@@ -0,0 +1,65 @@
+package com.apollographql.ijplugin.normalizedcache
+
+import com.apollographql.ijplugin.normalizedcache.NormalizedCache.Field
+import com.apollographql.ijplugin.normalizedcache.NormalizedCache.FieldValue
+import com.apollographql.ijplugin.normalizedcache.NormalizedCache.FieldValue.BooleanValue
+import com.apollographql.ijplugin.normalizedcache.NormalizedCache.FieldValue.CompositeValue
+import com.apollographql.ijplugin.normalizedcache.NormalizedCache.FieldValue.ListValue
+import com.apollographql.ijplugin.normalizedcache.NormalizedCache.FieldValue.Null
+import com.apollographql.ijplugin.normalizedcache.NormalizedCache.FieldValue.NumberValue
+import com.apollographql.ijplugin.normalizedcache.NormalizedCache.FieldValue.Reference
+import com.apollographql.ijplugin.normalizedcache.NormalizedCache.FieldValue.StringValue
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonNull
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.jsonObject
+import java.io.File
+import java.sql.DriverManager
+
+private const val REFERENCE_PREFIX = "ApolloCacheReference{"
```

Comment:
This duplicates some cache internals. We might want to expose a low level function in `apollo-normalized-cache` to help with this.

Reply:
I [ended up](https://github.com/apollographql/apollo-kotlin/pull/5306/commits/863e9c831979c765a9f198a00cbbce0a4a107328) using `apollo-normalized-cache-sqlite` instead of doing the db reading and json parsing manually.

Reply:
Even better 👍